doc/src/manual/multi-threading.md

```diff
@@ -213,3 +213,33 @@ therefore a blocking call like other Julia APIs.
 It is very important that the called function does not call back into Julia, as it will segfault.
 
 `@threadcall` may be removed/changed in future versions of Julia.
+
+## Caveats
+
+At this time, most operations in the Julia runtime and standard libraries
+are thread safe.
```

Comment:
```suggestion
are thread safe, except for `eval`.
```

Reply:
If we’re going to specifically call out one thing here, I think it should be the data structures. `eval` is covered by the top-level bullet below (but could probably be specifically called out).

Reply:
Right, the idea is for the subsequent list to list all the exceptions to this. I called out `include` because that will generally involve many kinds of top-level expressions and so probably isn't safe, but I don't think it's useful to say that `eval` is unsafe. Is `eval(:(1+2))` unsafe?

Reply:
I think the idea generally is that if it cant be done without eval, it’s unsafe right now. While that could have been a closure or something similar.

Reply:
Ok, that's a clearer rule. Maybe we could be even more specific and list some things like method definitions, global assignments, ... ?